### PR TITLE
implemented jwt authentication

### PR DESCRIPTION
**Pull request by annton-shakur**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstoreproject/controller/AuthenticationController.java [hunk 1]
```diff
@@ -20,6 +23,14 @@
 @RequestMapping("/api/auth")
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    @Operation(summary = "Login to a user account",
+            description = "Get a JWT token to a certain account")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/onlinebookstoreproject/controller/AuthenticationController.java]: Please add @Valid annotation to apply validation rules

File: src/main/java/mate/academy/onlinebookstoreproject/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.onlinebookstoreproject.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/onlinebookstoreproject/dto/user/UserLoginRequestDto.java:7]: Is it ok if email and password is string like "            "?

File: src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java]: Please don`t forget about constants

File: src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

File: src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/onlinebookstoreproject/security/JwtAuthenticationFilter.java]: magic number

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**